plugins/dell-dock/fu-dell-dock-usb-usb4.c

```diff
@@ -413,11 +417,19 @@ fu_dell_dock_usb4_hub_nvm_write (FuDevice *device, const guint8 *buf,
 static gboolean
 fu_dell_dock_usb4_activate (FuDevice *device, GError **error)
 {
+	g_autoptr(FuDeviceLocker) locker = fu_device_locker_new(device, error);
+	if (locker == NULL)
+		return FALSE;
+
 	if (!fu_dell_dock_usb4_hub_operation (device, OP_NVM_AUTH_WRITE,
 					      NULL, error)) {
 		g_prefix_error (error, "NVM authenticate failed: ");
 		return FALSE;
 	}
+
+	if (!fu_device_locker_close(locker, error))
+		return FALSE;
```

Comment:
I don't think you need to explicitly call close do you?  Shouldn't it happen automatically from using `g_autoptr`?

Reply:
> 1. when to do that
Will move it to _registered, thanks.

> 2. what happens to the EC's activation if you call USB4 online activation
EC's activation needs to wait until the dock cable is unplugged, no conflict for now but I will test it again when EC's activation for USB4 back normal. E.g. remove USB4 online activation.